monkey/infection_monkey/monkey.py

```diff
@@ -114,24 +115,22 @@ def __init__(self, args):
 
         self._singleton = SystemSingleton()
         self._opts = self._get_arguments(args)
-        self._server_strings = [str(s) for s in self._opts.servers]
 
         self._agent_event_serializer_registry = self._setup_agent_event_serializers()
 
         server, self._island_api_client = self._connect_to_island_api()
-        # TODO: `address_to_port()` should return the port as an integer.
-        self._cmd_island_ip, self._cmd_island_port = address_to_ip_port(server)
-        self._cmd_island_port = int(self._cmd_island_port)
+        self._cmd_island_ip = server.ip
+        self._cmd_island_port = server.port
 
         self._island_address = SocketAddress(
             IPv4Address(self._cmd_island_ip), self._cmd_island_port
         )
 
         self._control_client = ControlClient(
-            server_address=server, island_api_client=self._island_api_client
+            server_address=str(server), island_api_client=self._island_api_client
         )
-        self._control_channel = ControlChannel(server, get_agent_id(), self._island_api_client)
-        self._register_agent(server)
+        self._control_channel = ControlChannel(str(server), get_agent_id(), self._island_api_client)
```

Comment:
I'd say the same as above for ControlChannel but we're going to get rid of it soon so I don't think it's worth the effort.